cmd/gimlet/src/lib.rs

```diff
@@ -0,0 +1,151 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! ## `humility gimlet`
+//!
+//! `humility gimlet` contacts a (recent firmware version) Gimlet over the
+//! management network and extracts certain live diagnostic information.
+//!
+//! ```console
+//! $ humility gimlet --ip fe80::0c1d:9aff:fe64:b8c2%en0 read-seq-regs
+//! ```
+//!
+//! For a complete list of subcommands, use `humility gimlet --help`.
+
+use std::net::{SocketAddrV6, UdpSocket};
+use std::time::Duration;
+
+use anyhow::{bail, Context, Result};
+use clap::{ArgGroup, IntoApp, Parser};
+use humility::net::decode_iface;
+use humility_cli::{ExecutionContext, Subcommand};
+use humility_cmd::{Archive, Command, CommandKind, Dumper};
+
+/// This is defined in the gimlet TOML.
+const HARDCODED_PORT: u16 = 23547;
+
+#[derive(Parser, Debug)]
+#[clap(
+    name = "gimlet", about = env!("CARGO_PKG_DESCRIPTION"),
+    group = ArgGroup::new("target").multiple(false)
+)]
+struct Args {
+    /// How long to wait for a response from the target, in milliseconds.
+    #[clap(
+        long, short = 'T', default_value_t = 2000, value_name = "ms",
```

Comment:
TIL about `value_name`, that's cool!

Reply:
By default it uses your flag, uppercased, as the value name. I left this lowercase because the rest of Humility seems to use lowercase value names (manpages and Unix commandline help typically use uppercase value names).